src/main/java/org/openhds/service/impl/census/IndividualService.java

```diff
@@ -39,9 +71,45 @@ public Individual recordIndividual(Individual individual,
                                        String membershipUuid,
                                        String relationshipUuid,
                                        String fieldWorkerId) {
+
         individual.setCollectedBy(fieldWorkerService.findOrMakePlaceHolder(fieldWorkerId));
 
-        //TODO: Handle side effect creation for things like socialgroup and membership etc.
+        //Create the individual first, so that everything after is guaranteed to point at a real entity
+        individual = createOrUpdate(individual);
+
+        //TODO: Is the socialgroup created here?
+        //TODO: Do we make two paths depending on if it's a member or head of household?
```

Comment:
I think this method should have just one path.

Soon we will need different paths for member, head, and probably other use cases.  But I think these should be implemented in different methods.  That way we can avoid if-then-else logic, and we don't need to create an extra flag for choosing which branch.